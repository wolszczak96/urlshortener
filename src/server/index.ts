import { PrismaClient } from '@prisma/client'
import { createHash } from 'crypto'

export const db = new PrismaClient()

export function md5(text: string, slice = 6) {
  return createHash('md5').update(text).digest('hex').slice(0, slice)
}
