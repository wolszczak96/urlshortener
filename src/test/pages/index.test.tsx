import { render, fireEvent } from '../testUtils'
import Index from '@/pages'
import { waitFor } from '@testing-library/dom'
import '@testing-library/jest-dom'
import { LONG_URL, SHORT_URL, INVALID_URL } from '../constants'

import './__mocks__/server'

describe('Index page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Index />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button generates short url', async () => {
    const { getByPlaceholderText, getByText, getByDisplayValue } = render(
      <Index />,
    )

    // fill the input
    fireEvent.change(getByPlaceholderText('https://example.com'), {
      target: { value: LONG_URL },
    })
    await waitFor(() => {
      expect(getByDisplayValue(LONG_URL)).toBeInTheDocument()
    })

    // submit the form
    fireEvent.click(getByText('Shorten'))

    // dialog appears
    await waitFor(() => {
      expect(getByText('Hereâ€™s your shortened url:')).toBeInTheDocument()
    })
    await waitFor(() => {
      expect(getByDisplayValue(SHORT_URL)).toBeInTheDocument()
    })
  })

  it('validates input on submit', async () => {
    const { getByPlaceholderText, getByText, getByDisplayValue } = render(
      <Index />,
    )

    // fill the input
    fireEvent.change(getByPlaceholderText('https://example.com'), {
      target: { value: INVALID_URL },
    })
    await waitFor(() => {
      expect(getByDisplayValue(INVALID_URL)).toBeInTheDocument()
    })

    // submit the form
    fireEvent.click(getByText('Shorten'))

    // error appears
    await waitFor(() => {
      expect(getByText('Please enter a valid URL')).toBeInTheDocument()
    })
  })
})
