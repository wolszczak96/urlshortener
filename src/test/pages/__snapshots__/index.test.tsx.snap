// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="MuiBox-root MuiBox-root-1 css-b9gywi-StyledBox e1as48do1"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
    >
      Paste the URL you want to shorten
    </h6>
    <div
      class="MuiFormControl-root MuiTextField-root css-42wtm8-StyledInput e1as48do0"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="urlInput"
          placeholder="https://example.com"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-3"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Shorten
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
